optimize/util/optimize-commons/src/main/java/io/camunda/optimize/upgrade/os/OpenSearchClientBuilder.java

```diff
@@ -91,12 +96,31 @@ public class OpenSearchClientBuilder {
 
   public static ExtendedOpenSearchClient buildOpenSearchClientFromConfig(
       final ConfigurationService configurationService) {
-    return new ExtendedOpenSearchClient(buildOpenSearchTransport(configurationService));
+    final var pluginRepo = new PluginRepository();
```

Comment:
IMO better design would be to move PluginRepository instantiation outside of those client factory methods and pass it as a parameter, similar to how configurationService is passed. This should make OpenSearchClientBuilder better testable and less dependent on how PluginRepository is instantiated and initialized.

Reply:
That was my consideration too, but didn't do it in the first place because I was thinking that creating passing object as an argument in long chain might not have the best look. 

Do you suggest to create it in the [client factory create](https://github.com/camunda/camunda/blob/main/optimize/backend/src/main/java/io/camunda/optimize/service/db/os/OptimizeOpenSearchClientFactory.java#L32) or one level  higher in the `OptimizeOpenSearchClientConfiguration`?

Reply:
Looking into current design the best place to me seems to be io.camunda.optimize.service.db.os.OptimizeOpenSearchClientConfiguration#createOptimizeOpenSearchClient - this is the topmost place where all the dependencies are collected and passed as parameters. Yeah, I agree that passing dependencies as parameters in long chains is not the most convinient approach, but for static factory methods there is no other option, I suppose.

Reply:
Alright, I can do it